pkg/server/container_stop.go

```diff
@@ -154,21 +156,22 @@ func (c *criService) stopContainer(ctx context.Context, container containerstore
 	}
 
 	// Wait for a fixed timeout until container stop is observed by event monitor.
-	if err = c.waitContainerStop(ctx, container, killContainerTimeout); err == nil {
-		return nil
+	err = c.waitContainerStop(ctx, container)
+	if err != nil {
+		return errors.Wrapf(err, "an error occurs during waiting for container %q to be killed", id)
 	}
-	return errors.Wrapf(err, "an error occurs during waiting for container %q to be killed", id)
+	return nil
 }
 
-// waitContainerStop waits for container to be stopped until timeout exceeds or context is cancelled.
-func (c *criService) waitContainerStop(ctx context.Context, container containerstore.Container, timeout time.Duration) error {
-	timeoutTimer := time.NewTimer(timeout)
-	defer timeoutTimer.Stop()
+// waitContainerStop waits for container to be stopped until context is
+// cancelled or the context deadline is exceeded.
+func (c *criService) waitContainerStop(ctx context.Context, container containerstore.Container) error {
 	select {
 	case <-ctx.Done():
-		return errors.Wrapf(ctx.Err(), "wait container %q is cancelled", container.ID)
```

Comment:
Yea no problem. Was just keeping it the same as before in case a client was reading the text. I can make the update if you don't have concerns here?

Reply:
I see.

We can keep the message, but the error could still be ctx.Err(). :) Just in case new context errors added in the future.

Reply:
@Random-Liu - I think I misunderstood your question. Returning `ctx.Err()` will not suffice because gRPC does not automatically translate this to `codes.DeadlineExceeded`. So you still end up with: `rpc error: code = Unknown desc = context deadline exceeded` A client really needs `code` to be set appropriately. We could certainly add a handler in `instrumentedService` that tries to detect the `errors.Cause(err) == context.DeadlineExceeded` and translate it there but we cant just return `ctx.Err()` here.

Reply:
I see... My bad.